File: src/services/wishlist.service.js
import axiosClient from "@/api/axiosClient";

class WishlistService {
  // 1. Lấy danh sách yêu thích của sinh viên
  // Giả định API Backend: GET /api/wishlist/user/{studentId}
  async getWishlist(studentId) {
    return await axiosClient.get(`/wishlist/user/${studentId}`);
  }

  // 2. Thêm vào yêu thích
  // Giả định API Backend: POST /api/wishlist/add?studentId=...&courseId=...
  async addToWishlist(studentId, courseId) {
    return await axiosClient.post(`/wishlist/add`, null, {
      params: { studentId, courseId }
    });
  }

  // 3. Xóa khỏi yêu thích
  // Giả định API Backend: DELETE /api/wishlist/remove?studentId=...&courseId=...
  async removeFromWishlist(studentId, courseId) {
    return await axiosClient.delete(`/wishlist/remove`, {
      params: { studentId, courseId }
    });
  }

  // 4. Kiểm tra trạng thái (Có thể dùng nếu Backend có API check riêng)
  // Nếu không, ta sẽ check thủ công từ danh sách đã lấy
}

export default new WishlistService();

File: src/main/java/org/example/multileanproject/service/WishlistService.java
package org.example.multileanproject.service;

import org.example.multileanproject.dto.WishlistResponse; // Nhớ Import DTO
import java.util.List;

public interface WishlistService {
    // ✅ CHỈ GIỮ LẠI DÒNG NÀY (Trả về DTO)
    List<WishlistResponse> getWishlist(Long studentId);

    // ❌ XÓA DÒNG NÀY ĐI: List<Wishlist> getWishlist(Long studentId);

    void addToWishlist(Long studentId, Long courseId);
    void removeFromWishlist(Long studentId, Long courseId);
    void toggleWishlist(Long studentId, Long courseId);
}